resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,160 @@
+"""Prepare GEO."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import time
+import traceback
+
+from resdk.resources.utils import (
+    get_resource_collection, get_samples, get_resolwe
+)
+
+__all__ = ('prepare_geo_chipseq', 'prepare_geo_rnaseq', 'prepare_geo')
+
+
+def is_background(sample):
+    """Return ``True`` if given sample is background and ``False`` otherwise."""
+    background_relations = sample.resolwe.relation.filter(
+        type='compare',
+        label='background',
+        entity=sample.id,
+        position='background'
+    )
+    return len(background_relations) > 0
+
+
+def get_name_collection(collection_ids, res):
+    """Get the ``name`` input for ``prepare_geo`` processes."""
+    collection = None
+    if len(collection_ids) == 1:
+        collection_id = next(iter(collection_ids))
+        if collection_id:
+            collection = res.collection.get(collection_id)
+            name = collection.name
+        else:
+            name = time.strftime("%d-%m-%Y")
+    else:
+        name = time.strftime("%d-%m-%Y")
+    return name, collection
+
+
+def prepare_geo_chipseq(resource, name=None):
+    """Run ``Prepare GEO - ChIP-Seq`` process on the resource.
+
+    This method can be used to run ``Prepare GEO - ChIP-Seq`` process
+    on a single collection or a list of samples.
+
+    """
+
+    reads = []
+    macs14 = []
+    relations = []
+    samples = get_samples(resource)
+    resolwe = get_resolwe(*samples)
+    sample_ids = [sample.id for sample in samples]
+    collection_ids = set()
+    for sample in samples:
+        reads.append(sample.get_reads().id)
+        if is_background(sample):
+            continue
+
+        macs_list = sample.get_macs()
+        if macs_list:
```

Comment:
These three conditioncould be achieved in a single-level if-elif-else clause:

if not macs_list:
    ...
if len(macs_list) == 1:
    ...
else:
    ...